sql/src/main/java/io/crate/expression/tablefunctions/EmptyRowTableFunction.java

```diff
@@ -66,7 +61,9 @@ public FunctionInfo info() {
 
         @Override
         public RowType returnType() {
-            return emptyRow;
+            assert info.returnType() instanceof RowType
```

Comment:
Once https://github.com/crate/crate/pull/9853 is merged we should use the signature's return type. Probably `FunctionInfo` will be gone once all functions uses the signature registry.
Maybe we could wait a bit for #9853 as it maybe merged soon.

Reply:
👍 Sure, i'll rebase and adjust the PRs once #9853 is merged.

Reply:
As discussed, we can't do that atm as we'd need the bound signature for the correct return type. So we have to postpone this change until we may get rid of `FunctionInfo`.